### PR TITLE
Enable CLI logging

### PR DESCRIPTION
**Pull request by bukepo**
allow output logging through uart cli.

### CODE REVIEW

**Comment by bukepo** [comment]
This solves #541 and helps #590

**Comment by jwhui** [comment]
I like the idea of streaming the log through the same CLI output.  I think the logging functionality needs to be improved, but we can do that at a later time.  For now, how about the CLI class provides a method for printing the log output then define `otPlatLog()` such that it calls that newly defined method?

Also, I think the configure option should be named to indicate that it is tied to the CLI output (e.g. something like `--enable-cli-logging`.

**Comment by bukepo** [comment]
Got it, and cli-logging is much better. Thanks

File: src/cli/cli_uart.cpp [hunk 1]
```diff
@@ -205,13 +206,106 @@ extern "C" void otPlatUartSendDone(void)
...
-    mTxHead = (mTxHead + mSendLength) % kTxBufferSize;
-    mTxLength -= mSendLength;
-    mSendLength = 0;
+mTxHead = (mTxHead + mSendLength) % kTxBufferSize;
+mTxLength -= mSendLength;
+mSendLength = 0;
 
-    Send();
+Send();
 }
```
Comment (nibanks) [inline src/cli/cli_uart.cpp]: Accidental removal of the spaces?

Comment (jwhui) [inline src/cli/cli_uart.cpp]: It appears astyle may be tripping up on this.

@bukepo, try moving this method above the `otCliLog` declaration.

**Pull request merged**